Add modification to conda-based installation guide.

grmnptr opened the pull request:
Refs. #19452

## Reason
The guide is not clear on how the environmental variables are set. Blindly following the guide results in errors from not setting LIBMESH_DIR. Additionally, new mamba-based alternatives can be suggested by adding notes.

## Design
Modifying install_conda_moose.md

## Impact
Easier installation process for newcomers.

grmnptr committed:
Add modification to conda-based installation. (#19452)

loganharbour commented:
Waiting for @cticenhour to take a look as well

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/898829/) on f2c1c44 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/19453/site)

This comment will be updated on new commits.

cticenhour commented on modules/doc/content/getting_started/installation/install_conda_moose.md:
````diff
@@ -53,13 +53,46 @@ What ever is returned is your shell. With that information in hand (if you recei
+```
+mamba init
+```
+Following this, close the terminal, open a new one and activate the environment using:
````
```suggestion

Following this, close the terminal, open a new one and activate the environment using:

```
Add spaces around this to properly format the code snippets within the note.

cticenhour reviewed (changes requested):
Only one change request related to formatting. Thank you for adding this! 😄

grmnptr committed:
Add Casey's modification. (#19452)

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/899063/) on 6bb1699 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/tensor_mechanics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/19453/coverage/xfem)

</details>



This comment will be updated on new commits.

milljm commented:
I like this change. But I don't like the combinations I am seeing with miniconda/mamba.

I suggest we completely change the links where folks are grabbing their conda installer. i.e. no longer send them to https://github.com/mamba-org/mamba (which only instructs them to `conda install mamba`, and instead send them to https://github.com/conda-forge/miniforge, to download one of the following:

- Linux
  - https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh
- Macintosh Intel
  - https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-MacOSX-x86_64.sh
- Macintosh ARM
  - https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-MacOSX-arm64.sh

We need to move away from Miniconda entirely, as the licensing with Anaconda (pkg/main channel) is a bit up in the air with open source, and due to the fact it does not support ARM anyway.


I am talking about a change to this page: https://mooseframework.inl.gov/getting_started/installation/conda.html

If we use Mambaforge, we can drop explaining to folks how to add `conda-forge` channel as well. Basically, by switching to Mambaforge installers, we will be simplifying the Install page and start officially supporting ARM.

milljm commented:
I'll leave it up to you to change the installation path :)

~/mambaforge3? ~/conda3? ~/conda?  🤷  whatever you guys want.

grmnptr committed:
Rewriting documentation to use mambaforge/mamba. (#19452)

loganharbour commented on modules/doc/content/getting_started/installation/install_mambaforge.md:
```diff
@@ -0,0 +1,41 @@
+## Install Mambaforge3 id=installconda
+
+Installing Mambaforge3 is straight forward. Download, install, and configure. If you run into issues during these steps, please visit our [troubleshooting guide for Conda](troubleshooting.md#condaissues optional=True). This installation
+guide relies on the utilization of `mamba`, a C-based package manager for Conda that takes advantage of parallelization
+and is, therefore, considerable faster.   
```
```suggestion
Follow the steps below depending on your platform to install mambaforge. If you run into issues during these steps, please visit our [troubleshooting guide for Conda](troubleshooting.md#condaissues optional=True). This installation
guide relies on the utilization of `mamba`, an optimized package manager for Conda.
```

cticenhour replied:
I like this change.

cticenhour replied:
I like this change.

milljm commented on modules/doc/content/getting_started/installation/conda.md:
```diff
@@ -7,7 +7,7 @@ an option to install MOOSE directly on a Windows system is not yet supported.
 and an experimental [WSL](installation/windows10.md) option is available.
 
-!include installation/install_miniconda.md
+!include installation/install_mambaforge.md
```
Because of the current error for "[Moose Install Target Test](https://civet.inl.gov/job/900508/)" I am wondering if we should rename this to simply 'install_conda.md'.

[github.com mastodon mastodon_conda.md](https://github.com/idaholab/mastodon/blob/650ffff487934ed6215cb2dc11124910a3873dee/doc/content/getting_started/mastodon_conda.md?plain=1#L17)

So we don't run into this again when/if we happen to switch to some other conda installer. There might be other application documentations which is referencing this file name.

cticenhour commented on modules/doc/content/getting_started/installation/install_conda_moose.md:
```diff
@@ -1,66 +1,36 @@
-See 'conda init --help' for more information and options.
-
-IMPORTANT: You may need to close and restart your shell after running 'conda init'.
+mamba init
```
So we do not need to specify the shell for the `mamba init` command as we did for regular conda?

grmnptr replied:
Yes, the mamba script finds the shell.

cticenhour commented on modules/doc/content/getting_started/installation/install_mambaforge.md:
````diff
@@ -0,0 +1,41 @@
+  bash Mambaforge-MacOSX-x86_64.sh -b -p ~/mambaforge3
+  ```
+
+- +Macintosh Users with ARM processors:+
+
+  ```bash
+  curl -L -O https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-MacOSX-arm64.sh
+  bash Mambaforge-MacOSX-arm64.sh -b -p ~/mambaforge3
+  ```
````
Since we don't officially support ARM platforms on Mac (yet), I think this should be removed until we do.

milljm replied:
darn... yeah.. . sad panda

cticenhour commented on modules/doc/content/getting_started/installation/install_mambaforge.md:
````diff
@@ -0,0 +1,41 @@
+```
+
+!alert warning title=sudo conda
+If you find yourself using `sudo conda`/`sudo mamba`... something is not right. The most common reason for needing sudo, is due to an improper Conda installation. Conda *should* be installed to your home directory, without any use of `sudo`.
````
```suggestion
If you find yourself using `sudo conda`/`sudo mamba`... something is not right. The most common reason for needing sudo is due to an improper Conda installation. Conda *should* be installed to your home directory, without any use of `sudo`.
```

cticenhour reviewed (dismissed):
Thank you for re-writing these instructions! Just a few comments.

grmnptr committed:
Apply suggestions from code review. (#19452)

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>
Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

grmnptr committed:
Add more suggestions from code review. (#19452)

milljm commented:
Yes huge thank you for doing this @grmnptr !

cticenhour commented:
@grmnptr can you rebase on current devel? No idea why you're getting tensor mechanics failures here.

loganharbour commented:
> @grmnptr can you rebase on current devel? No idea why you're getting tensor mechanics failures here.

No need. I pinged @milljm about this yesterday. I bet the failing Mac is the one with the new OS on it. Someone found this on another PR but I can’t remember where it is.

cticenhour commented:
I don't remember seeing this on Monterey on my laptop with the newest configurations. Were you guys able to replicate?

loganharbour commented:
I haven’t looked into it at all but I will today

milljm commented:
going to work on this... (mastodon update). I really like these new instructions.

milljm commented:
I think the best path forward may be to create symlinks to the older counterpart install markdown files. Commit, then fix the other apps which may be broken (like Mastodon)

Commit:
Create symlink to older install_miniconda.md

In order to not break other Apps, create symlink to miniconda instructions.
Aftwards, clean up other apps which may still be using this file to reference
the proper one.

Refs #19452

Commit:
Revert back to name: install_miniconda.md

Due to an issue beyond my knowledge, a symlink doesn't work for a specific application.
Change the naming convention back to the original method for now :(

loganharbour commented on modules/doc/content/getting_started/installation/install_conda_moose.md:
````diff
@@ -1,66 +1,36 @@
 ```
 
-This error is trying to exlain how to initialize Conda for the first time for your given shell environment. Different operating systems use different shells. To understand what shell you are using, perform the following:
+Next, we create a moose environment, and attempt to activate it:
````
Next, create a unique conda environment for moose, named `moose`, and attempt to activate it:

loganharbour commented on modules/doc/content/getting_started/installation/install_conda_moose.md:
````diff
@@ -1,66 +1,36 @@
 ```
 
-What ever is returned is your shell. With that information in hand (if you received the above error to begin with) perform the following init command:
+As a next step, the packages necessary for compiling/running moose are installed.
````
Within the `moose` environment, install the necessary packages:

loganharbour commented on modules/doc/content/getting_started/installation/install_conda_moose.md:
````diff
@@ -1,66 +1,36 @@
 ```
 
-Where SHELL_NAME is the shell you discovered in the previous step. Once complete, close any terminals you have opened, and re-open them. Then run `conda activate moose` again. If successful proceed to installing packages:
+Once the packages are installed, the environment needs to be deactivated and
````
```suggestion
Once the packages are installed, the `moose` environment needs to be deactivated and
```

Commit:
Address requested changes

To more clear about the conda environment name.
